### PR TITLE
[NO-JIRA] Change Fab base class to FloatingActionButton

### PR DESCRIPTION
**Pull request by tiagoengel**
The app is using a few features of `FloatingActionButton` that we do not provide, which means they cannot use this component atm. It's easier to change the base classe for now to help with adoption and think about sharing code (between button and fab) later.

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/src/main/java/net/skyscanner/backpack/fab/BpkFab.kt [hunk 1]
```diff
@@ -4,28 +4,22 @@ import android.animation.AnimatorInflater
...
 import net.skyscanner.backpack.util.createContextThemeWrapper
-import net.skyscanner.backpack.util.darken
 import net.skyscanner.backpack.util.use
 
 open class BpkFab @JvmOverloads constructor(
   context: Context,
   attrs: AttributeSet? = null,
   defStyleAttr: Int = 0
-) : AppCompatButton(createContextThemeWrapper(context, attrs, R.attr.bpkFabStyle), attrs, defStyleAttr) {
+) : FloatingActionButton(createContextThemeWrapper(context, attrs, R.attr.bpkFabStyle), attrs, defStyleAttr) {
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/fab/BpkFab.kt]: Please wrap it into com.google.android.material.R.attr.floatingActionButtonStyle as well.

File: Backpack/src/main/java/net/skyscanner/backpack/fab/BpkFab.kt [hunk 2]
```diff
@@ -4,28 +4,22 @@ import android.animation.AnimatorInflater
...
 import androidx.core.content.ContextCompat
 import androidx.core.graphics.drawable.DrawableCompat
+import com.google.android.material.floatingactionbutton.FloatingActionButton
 import net.skyscanner.backpack.R
 import net.skyscanner.backpack.util.BpkTheme
 import net.skyscanner.backpack.util.createContextThemeWrapper
-import net.skyscanner.backpack.util.darken
 import net.skyscanner.backpack.util.use
 
 open class BpkFab @JvmOverloads constructor(
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/fab/BpkFab.kt]: Please delegate the attributes to FloatingActionButton implementation.

Comment (tiagoengel) [reply Backpack/src/main/java/net/skyscanner/backpack/fab/BpkFab.kt]: 👍

File: Backpack/src/main/java/net/skyscanner/backpack/fab/BpkFab.kt (hunk 1 above)
Comment (tiagoengel) [reply Backpack/src/main/java/net/skyscanner/backpack/fab/BpkFab.kt]: 👍

**Pull request merged**